### PR TITLE
Restructure project layout - split infrastructure and opinionated project generation into separate subprojects

### PR DESCRIPTION
**Pull request by tgeens**

### CODE REVIEW

File: start.xenit.eu/build.gradle [hunk 1]
```diff
@@ -0,0 +1,70 @@
+plugins {
+    id 'java'
+    id 'org.springframework.boot'
+    id 'io.spring.dependency-management'
+    id 'eu.xenit.docker' version '4.0.3'
+    id 'idea'
+}
+
+sourceCompatibility = 1.8
```
Comment (driesdeproost) [inline start.xenit.eu/build.gradle]: why is there a sourceCompatibility but not a targetCompatibility?

Comment (tgeens) [reply start.xenit.eu/build.gradle]: `targetCompatibility` defaults to the value of `sourceCompatibility`

https://docs.gradle.org/current/userguide/java_plugin.html#sec:java_convention_properties

File: initializr-generator/build.gradle [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+repositories {
+    mavenCentral()
+
+    maven { url "https://repo.spring.io/snapshot" }
+    maven { url "https://repo.spring.io/milestone" }
+    maven { url "https://repo.spring.io/release" }
+}
+
+dependencies {
+//    implementation('org.springframework.boot:spring-boot-starter-web')
```
Comment (driesdeproost) [inline initializr-generator/build.gradle]: comments still needed?

Comment (tgeens) [reply initializr-generator/build.gradle]: cleaned up

**Pull request merged**